kernel/src/main/java/org/kframework/parser/ParserUtils.java

```diff
@@ -202,15 +202,15 @@ public Set<Module> loadModules(
 
         new CollectProductionsVisitor(kore, context).visit(def);
 
-        // Tuple3 of moduleName, Source, Location
-        Map<String, List<Tuple3<String, Source, Location>>> groupedModules =
+        // Tuple3 of moduleName, Source, Location, digest
+        Map<String, List<Tuple4<String, Source, Location, Long>>> groupedModules =
```

Comment:
this is probably not what you want. The tuple of String, Source, Location was being used to identify modules that appeared in two places with different filenames or locations. You probably just instead want to identify two modules that have the same name but different contents, in which case you really only need module name and digest in the value of the map.